cache-config/t3cutil/toreq/clientfuncs.go

```diff
@@ -720,7 +720,7 @@ func (cl *TOClient) GetServerUpdateStatus(cacheHostName tc.CacheName, reqHdr htt
 		if err != nil {
 			return errors.New("getting server update status from Traffic Ops '" + torequtil.MaybeIPStr(reqInf.RemoteAddr) + "': " + err.Error())
 		}
-		status := obj.(*tc.ServerUpdateStatus)
+		status := obj.(*tc.ServerUpdateStatusV4)
```

Comment:
This line is causing the segmentation fault - t3c has two client libraries to handle the two API versions it consumes. I'm not extremely familiar with the architecture, but I'm fairly certain you'll need to be changing this in at least two places - and one of those will involve coercing the older structure to the newer.